crates/compiler/types/src/types.rs

```diff
@@ -3420,6 +3622,13 @@ pub struct RecordStructure {
     pub ext: Variable,
 }
 
+#[derive(Debug)]
+pub struct TupleStructure {
+    /// Invariant: these should be sorted!
+    pub elems: Vec<(usize, Variable)>,
```

Comment:
there is a data structure VecMap in roc_collections that might be better suited here to help ensure order and no duplicates